app/components/tracks/TracksController.js

```diff
@@ -31,6 +33,65 @@ tracksModule.controller('TracksController',
                 }, function (error) {
                     console.error(error);
                 });
+
+            ApiJsonFactory.getJson('sessions')
+                .then(function (response) {
+                    tc.Sessions = response.data.sessions;
+                    $sessionStorage.sessions = tc.Sessions;
+                }, function (error) {
+                    console.error(error);
+                });
         }
 
-}]);
\ No newline at end of file
+        tc.showSession = function(track, event) {
```

Comment:
yeah actually how are you defining `singleTrack` ? You should consider passing it as a dependency to the controller instead of relying on the global scope of things.